haigha/transports/socket_transport.py

```diff
@@ -29,26 +34,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
         self._host = "%s:%s" % (host, port)
-        self._sock = klass()
-        self._sock.setblocking(True)
-        self._sock.settimeout(self.connection._connect_timeout)
-        if self.connection._sock_opts:
-            for k, v in self.connection._sock_opts.iteritems():
-                family, type = k
-                self._sock.setsockopt(family, type, v)
-        infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
-        exc = socket.error("getaddrinfo returns an empty list")
-        for _family, _socktype, _proto, _canonname, sockaddr in infos:
+
+        for info in socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP):
```

Comment:
not sure that we need to raise a custom error though, the stdlib exceptions do a fine job here and are easily handled by the caller. I think in all cases, the connect is directly called by the user through `Connection.connect`, and if that is documented to raise socket exceptions then I think we should stick with that.